### PR TITLE
Made 75%+ of the panel translatable

### PR DESCRIPTION
**Pull request by Miniontoby**
Made 75%+ of the panel translatable

The rest will come soon, but it is too hard to keep up!

Includes changes to:
- Filament code
- Original ptero frontend
- Original ptero admin view
- languages files, EN and NL

To discuss, you could also use https://github.com/pelican-dev/panel/discussions/79

### CODE REVIEW

**Comment by Miniontoby** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Miniontoby** [comment]
Already translated it back when there was only ptero, so I just added it... but it is fine

**Comment by Miniontoby** [comment]
Pull request is ready to be merged, if approved of course.

The missing things will be added in a later commit.
I could/*would like to* take the task to make everything translatable on my name, since it is the best to do let one person do it all... but yeah, up to you guys.

But first merge the request if it is oké.

File: lang/nl/admin/index.php
Comment (Fabian-Ser) [inline lang/nl/admin/index.php:1]: Please look to 'not-up-to-date2':
'and jij draait op dit moment versie <code>:version</code>.', -> 'en jij draait op dit moment versie <code>:version</code>.',

File: lang/nl/admin/layout.php
Comment (Fabian-Ser) [inline lang/nl/admin/layout.php:1]: Toggle navigation is not translated properly, Admin Control should be 'Admin Beheer'

File: lang/nl/admin/navigation.php
Comment (Fabian-Ser) [inline lang/nl/admin/navigation.php:1]: Databasen -> Databases

File: lang/nl/dashboard/account.php
Comment (Fabian-Ser) [inline lang/nl/dashboard/account.php:1]: Update -> Wijzigen (verb)

File: lang/nl/dashboard/account.php [hunk 1]
```diff
@@ -1,28 +1,48 @@
...
+            'field' => 'Voer token in',
+            'button' => 'Schakel Twee-Staps uit',
+        ],
+        'setup' => [
+            'title' => 'Schakel Twee-Staps Verificatie in',
+            'subtitle' => "Help protect your account from unauthorized access. You'll be prompted for a verification code each time you sign in.",
+            'help' => 'Scan de bovenstaande QR-code met de tweestapsauthenticatie-app van uw keuze. Voer vervolgens de gegenereerde zescijferige code in het onderstaande veld in.',
+        ],
+
+        'required' => [
```
Comment (Fabian-Ser) [inline lang/nl/dashboard/account.php:43]: This is not translated

File: lang/nl/dashboard/account.php [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
     'two_factor' => [
-        'button' => 'Tweestapsverificatie configureren',
-        'disabled' => 'Tweestapsverificatie is uitgeschakeld voor je account. Je wordt niet meer gevraagd om een code op te geven bij het inloggen.',
-        'enabled' => 'Tweestapsverificatie is ingeschakeld op je account! Vanaf nu moet je bij het inloggen de code opgeven die door je apparaat wordt gegenereerd.',
-        'invalid' => 'De opgegeven code is ongeldig.',
-        'setup' => [
-            'title' => 'Tweestapsverificatie instellen',
-            'help' => 'Kan de code niet worden gescand? Voer de onderstaande code in je applicatie:',
-            'field' => 'Code invoeren',
+        'title' => 'Twee-Staps Verificatie',
```
Comment (Fabian-Ser) [inline lang/nl/dashboard/account.php:22]: Tweestapsverificatie is one word

File: lang/nl/strings.php [hunk 3]
```diff
@@ -3,16 +3,34 @@
...
+    'mounts' => 'Mount|Mounts',
+    'eggs' => 'Egg|Eggs',
+    'nodes' => 'Node|Nodes',
+    'servers' => 'Server|Servers',
+    'users' => 'Gebruiker|Gebruikers',
+    'force_outgoing_ip' => 'Forceer Uitgaande IP',
+    'startup' => 'Startup',
+    'docker_images' => 'Docker Images',
+    'linked_node' => 'Gekoppelde Node',
+    'primary_allocation' => 'Primaire Allocatie',
```
Comment (Fabian-Ser) [inline lang/nl/strings.php:33]: Allocatie -> Toewijzing to be consistent with current lang files

File: lang/nl/strings.php [hunk 4]
```diff
@@ -31,19 +51,17 @@
...
     'sftp' => 'SFTP',
-    'databases' => 'Databases',
     'memo' => 'Notitie',
     'created' => 'Aangemaakt',
     'expires' => 'Vervalt',
     'public_key' => 'Token',
-    'api_access' => 'Api toegang',
-    'never' => 'nooit',
-    'sign_out' => 'Afmelden',
+    'api_access' => 'Api Toegang',
```
Comment (Fabian-Ser) [inline lang/nl/strings.php:58]: API

**Comment by PascaleBeier** [comment]
Is there a reason you chose trans over __()? The latter gives much more flexibility for JSON translation and context substitution

File: lang/nl/strings.php [hunk 5]
```diff
@@ -58,21 +76,25 @@
     'action' => 'Actie',
     'data' => 'Gegevens',
     'queued' => 'In wachtrij',
-    'last_run' => 'Laatst uitgevoerd',
-    'next_run' => 'Volgende uitvoering',
-    'not_run_yet' => 'Nog niet uitgevoerd',
+    'last_run' => 'Laatst Uitgevoerd',
```
Comment (Fabian-Ser) [inline lang/nl/strings.php:79]: Please be consistent in your capitalisation,

File: lang/nl/strings.php [hunk 6]
```diff
@@ -58,21 +76,25 @@
...
+    'next_run' => 'Volgende Uitvoering',
+    'not_run_yet' => 'Nog Niet Uitgevoerd',
     'yes' => 'Ja',
     'no' => 'Nee',
     'delete' => 'Verwijderen',
-    '2fa' => 'Tweestapsverificatie',
+    '2fa' => '2FA',
```
Comment (Fabian-Ser) [inline lang/nl/strings.php:85]: Write it out to Tweestapsverificatie to be consistent with errors, buttons etc.

File: lang/nl/strings.php [hunk 7]
```diff
@@ -58,21 +76,25 @@
...
-    'read_only' => 'Alleen lezen',
+    'read_only' => 'Alleen Lezen',
+    'read_only?' => 'Alleen Lezen?',
+    'writable' => 'Schrijfbaar',
     'relation' => 'Relatie',
     'owner' => 'Eigenaar',
     'admin' => 'Beheerder',
     'subuser' => 'Medegebruiker',
+    'author' => 'Schrijver',
+    'image_uri' => 'Image URI',
```
Comment (Fabian-Ser) [inline lang/nl/strings.php:97]: Translation missing

**Review by Fabian-Ser — COMMENTED** [review]
Please work on the Dutch translations

File: lang/nl/strings.php (hunk 6 above)
Comment (Miniontoby) [reply lang/nl/strings.php:85]: This isn't supposed to be a translating PR, tho. I specifically left things out UNTRANSLATED to indicate that they AREN'T used.

File: lang/nl/strings.php (hunk 5 above)
Comment (Miniontoby) [reply lang/nl/strings.php:79]: It should be capitalized, tho. Since the original one was also

File: lang/nl/strings.php (hunk 4 above)
Comment (Miniontoby) [reply lang/nl/strings.php:58]: Api was translated as Api by someone else

File: lang/nl/strings.php (hunk 6 above)
Comment (Fabian-Ser) [reply lang/nl/strings.php:85]: Why should you not remove them if they are not being used?

**Comment by Miniontoby** [comment]
@fabian-ser

This isn't supposed to be a translating/translation PR, tho. I specifically left things out UNTRANSLATED to indicate that they AREN'T used.

File: lang/nl/admin/index.php
Comment (Miniontoby) [reply lang/nl/admin/index.php:1]: Yeah, someone changed the strings when i already did that one and then I forgot to update it, but again:

This isn't supposed to be a translating/translation PR, tho.

**Review by Poseidon281 — COMMENTED** [review]
The code itself is looking good. I am not getting any errors. @lancepioch I think this PR can be merged and once it is merged, I will go through it and translate the rest of the code and create files for it etc..

**Comment by Miniontoby** [comment]
The dutch translations that I added are used, since that is the way how I checked if it worked, but it is better to translate using Crowdin instead of using `nano`. 

Translations can be fixed by others afterwards. But I wanted to see what files are in english folder, but unused, so that translators (also me for some parts, still) do not translate more than needed

File: lang/nl/strings.php (hunk 6 above)
Comment (Miniontoby) [reply lang/nl/strings.php:85]: 2FA is fine in dutch, tho

File: lang/nl/strings.php (hunk 7 above)
Comment (Miniontoby) [reply lang/nl/strings.php:97]: Image is Docker image, cannot translate that to Afbeelding.

URI is also URI, afaik

Comment (Miniontoby) [reply lang/nl/strings.php:97]: Image is Docker image, cannot translate that to Afbeelding.

URI is also URI, afaik

File: lang/nl/strings.php (hunk 3 above)
Comment (Miniontoby) [reply lang/nl/strings.php:33]: I haven't looked at the current lang file, since it wasn't in my folder when I started working on this PR

**Comment by Miniontoby** [comment]
> @Miniontoby Clear, I would say keep only the English translation (and remove the unused) as these translations could be used to translate it to the other languages via Crowdin. And remove the Dutch translations as these contains unused variables and grammar/consistency errors.

Problem with removing the unused English files, is that this PR is not fully finished.

However, I am not going to put this one on hold, since it already takes me too much time to keep up-to-date.

I don't know if I use a bad merge option or something, but at least it always complains and I just spend too much time into fixing my .patch file.

I will remove the Dutch translations, as you wish...

File: lang/nl/admin/databases.php [hunk 8]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+return [
+    'title' => 'Database Hosts',
+    'header' => [
+        'title' => 'Database Hosts',
+        'subtitle' => 'Database hosts waar servers databasen op kunnen maken.',
+    ],
+    'content' => [
+    ],
```
Comment (Fabian-Ser) [inline lang/nl/admin/databases.php:10]: This content array is empty

**Comment by Poseidon281** [comment]
> @Miniontoby Clear, I would say keep only the English translation (and remove the unused) as these translations could be used to translate it to the other languages via Crowdin. And remove the Dutch translations as these contains unused variables and grammar/consistency errors.

I will go over all the translations anyway before uploading them to Crowdin, but this PR is just to see if the code works. I go over the translations both English and Dutch before they officially go to the Crowdin.

File: lang/nl/admin/databases.php (hunk 8 above)
Comment (Miniontoby) [reply lang/nl/admin/databases.php:10]: Yes that is true, that one needs to be added, but this is from the Ptero /admin view.
When I saw the new filament branch, I stopped working on the /admin view. but this should stay

File: lang/nl/strings.php [hunk 9]
```diff
@@ -91,5 +113,15 @@
     'enable' => 'Inschakelen',
     'disable' => 'Uitschakelen',
     'save' => 'Opslaan',
-    'copyright' => '&reg; 2024 - :year Pelican',
+    'error' => 'Fout',
+    'error_rendering_view' => 'Er is een fout opgetreden in de toepassing tijdens het weergeven van deze weergave. Probeer de pagina te vernieuwen.',
+    'okay' => 'Oke',
+    'pterodactyl_icon' => 'Pterodactyl Icoon',
```
[the remark this answers is not in the document]
Comment (Miniontoby) [reply lang/nl/strings.php:119]: `pterodactyl_icon` is indeed not being used, yeah

**Comment by Miniontoby** [comment]
The problem is just that this repo is too active, (which I do not disagree on) but that just gives me less time to translate my stuff correctly. I am able to translate to Dutch correctly!

I do not want to have a low reputation! I am fully capable of translating Dutch, but yeah, atm I just have to put 10 times MORE time into keeping my patch up-to-date, which I would have wanted to put into translating the keys correctly!

**Comment by Poseidon281** [comment]
> The problem is just that this repo is too active, (which I do not disagree on) but that just gives me less time to translate my stuff correctly. I am able to translate to Dutch correctly!
> 
> I do not want to have a low reputation! I am fully capable of translating Dutch, but yeah, atm I just have to put 10 times MORE time into keeping my patch up-to-date, which I would have wanted to put into translating the keys correctly!
> 
> I have already put 20 days (20 * 24 hrs) into making this stupid little PR! One more comment or review that indicates any mishaps or whatever and I will just cancel my PR, likely

This Project only exists for 10 days?

**Comment by Miniontoby** [comment]
> > The problem is just that this repo is too active, (which I do not disagree on) but that just gives me less time to translate my stuff correctly. I am able to translate to Dutch correctly!
> > I do not want to have a low reputation! I am fully capable of translating Dutch, but yeah, atm I just have to put 10 times MORE time into keeping my patch up-to-date, which I would have wanted to put into translating the keys correctly!
> > I have already put 20 days (20 * 24 hrs) into making this stupid little PR! One more comment or review that indicates any mishaps or whatever and I will just cancel my PR, likely
> 
> This Project only exists for 10 days?

I began LONG before Pelican. I already started at Pterodactyl!

**Comment by Miniontoby** [comment]
> > The problem is just that this repo is too active, (which I do not disagree on) but that just gives me less time to translate my stuff correctly. I am able to translate to Dutch correctly!
> > I do not want to have a low reputation! I am fully capable of translating Dutch, but yeah, atm I just have to put 10 times MORE time into keeping my patch up-to-date, which I would have wanted to put into translating the keys correctly!
> > I have already put 20 days (20 * 24 hrs) into making this stupid little PR! One more comment or review that indicates any mishaps or whatever and I will just cancel my PR, likely
> 
> This Project only exists for 10 days?

I began LONG before Pelican. I already started at Pterodactyl!

File: lang/en/strings.php [hunk 10]
```diff
@@ -91,5 +113,14 @@
     'enable' => 'Enable',
     'disable' => 'Disable',
     'save' => 'Save',
-    'copyright' => '&reg; 2024 - :year Pelican',
```
Comment (Poseidon281) [inline lang/en/strings.php:94]: Why did you delete this?

**Comment by Boy132** [comment]
Maybe it would be better to wait for a release and _then_ do the translations?

**Comment by Poseidon281** [comment]
> Maybe it would be better to wait for a release and _then_ do the translations?

Yes that is a good idea. A lot of code is still changing now, so it is also inconvenient to do it now.

**Comment by lancepioch** [comment]
Awesome, thank you a ton for the contribution! I'll try to take a look at this tonight or tomorrow night. I've got two notes for you:

1. I could be wrong, but I think it doesn't make sense for most languages to actually translate the resource names, such as Egg, Node, Server, etc. They have a specific and distinct meaning in our app that wouldn't be synonymous if translated 1:1. The descriptions of them would be a great place for translations though.
2. Also could you take your language specific changes out as well and submit them on Crowdin instead?

**Comment by Miniontoby** [comment]
> Maybe it would be better to wait for a release and then do the translations?

No, then you are way behind!


> Awesome, thank you a ton for the contribution! I'll try to take a look at this tonight or tomorrow night. I've got two notes for you:
>
> 1. I could be wrong, but I think it doesn't make sense for most languages to actually translate the resource names, such as Egg, Node, Server, etc. They have a specific and distinct meaning in our app that wouldn't be synonymous if translated 1:1. The descriptions of them would be a great place for translations though.
> 2. Also could you take your language specific changes out as well and submit them on Crowdin instead?

1. Yeah, but I thought with like plurals, it might be different..
2. I will remove the Dutch translation files, that I created when there was the `feature/filament` branch that didn't include ANY translation files

You can expect it to be done soon.


And one note. If there will be too much delay or too much extra commentary about my changes, I might be ABORTing this pull request for good...

**Comment by Miniontoby** [comment]
Oke, reverted the dutch translation changes.

About the project specific terms, I don't know, but yeah, they were already in the strings.php, so you cannot complain to me, about the fact that those shouldn't be translated, I guess

**Comment by Poseidon281** [comment]
> Oke, reverted the dutch translation changes.
> 
> About the project specific terms, I don't know, but yeah, they were already in the strings.php, so you cannot complain to me, about the fact that those shouldn't be translated, I guess

It is not that we do not appreciate that you have also made your files in Dutch, because we certainly do, but it is easier for us to update and sync Crowdin if all languages except English are only done through Crowdin. Also this way it is also easier to oversee for other translators and developers. That is why I would like to ask you to remove the Dutch translations and upload them to Crowdin if your PR is merged and everything is synced.

**Comment by Miniontoby** [comment]
Yeah, I do understand the reasons.
I do still have a copy of the dutch translations, but of course I cannot upload them myself.

Btw with the crowdin-cli you can easily upload sources and translations! Just so you know.

But yeah I guess we do it in the way that the 'masters' say.

File: app/Filament/Resources/ApiKeyResource.php [hunk 11]
```diff
@@ -11,7 +11,6 @@
 class ApiKeyResource extends Resource
 {
     protected static ?string $model = ApiKey::class;
```
Comment (lancepioch) [inline app/Filament/Resources/ApiKeyResource.php:13]:
```suggestion
    protected static ?string $model = ApiKey::class;
    protected static ?string $label = 'API Key';
```

File: app/Filament/Resources/UserResource/Pages/EditProfile.php [hunk 12]
```diff
@@ -37,10 +37,11 @@ protected function getForms(): array
                     ->schema([
                         Tabs::make()->persistTabInQueryString()
                             ->schema([
-                                Tab::make('Account')
+                                Tab::make(trans('strings.account'))
```
Comment (lancepioch) [inline app/Filament/Resources/UserResource/Pages/EditProfile.php:40]:
```suggestion
                                Tab::make('Account')->label(trans('strings.account'))
```

File: routes/api-client.php [hunk 13]
```diff
@@ -29,6 +29,7 @@
 
     Route::put('/email', [Client\AccountController::class, 'updateEmail'])->name('api:client.account.update-email');
     Route::put('/password', [Client\AccountController::class, 'updatePassword'])->name('api:client.account.update-password');
+    Route::get('/language', [Client\AccountController::class, 'getLocale']);
```
Comment (lancepioch) [inline routes/api-client.php:32]:
```suggestion
```

File: app/Http/Controllers/Api/Client/AccountController.php [hunk 14]
```diff
@@ -72,4 +72,14 @@ public function updatePassword(UpdatePasswordRequest $request): JsonResponse
...
+
+    /**
+     * Get user language
+     */
+    public function getLocale(Request $request): JsonResponse
+    {
+        $original = $request->user()->language;
+
+        return new JsonResponse($original);
+    }
```
Comment (lancepioch) [inline app/Http/Controllers/Api/Client/AccountController.php:84]:
```suggestion
```

File: app/Providers/Filament/AdminPanelProvider.php [hunk 15]
```diff
@@ -3,6 +3,7 @@
 namespace App\Providers\Filament;
 
 use App\Filament\Resources\UserResource\Pages\EditProfile;
+use App\Http\Middleware\LanguageMiddleware;
```
Comment (lancepioch) [inline app/Providers/Filament/AdminPanelProvider.php:6]:
```suggestion
```

File: app/Providers/Filament/AdminPanelProvider.php [hunk 16]
```diff
@@ -67,6 +68,7 @@ public function panel(Panel $panel): Panel
                 SubstituteBindings::class,
                 DisableBladeIconComponents::class,
                 DispatchServingFilamentEvent::class,
+                LanguageMiddleware::class,
```
Comment (lancepioch) [inline app/Providers/Filament/AdminPanelProvider.php:71]:
```suggestion
```

**Comment by Miniontoby** [comment]
What happened! Why are there now just 4 files left!

**Comment by Miniontoby** [comment]
Oh I see, parts of it are already merged...

And I understand the reason to remove the Panel related words, but I believe Users is not a panel related word:
https://github.com/pelican-dev/panel/blob/51bb60c3b12ab5ad4095447d8cc868bd039444ac/resources/views/filament/pages/dashboard.blade.php#L30

File: app/Filament/Resources/UserResource.php [hunk 17]
```diff
@@ -17,11 +17,6 @@ class UserResource extends Resource
 {
     protected static ?string $model = User::class;
 
-    public static function getLabel(): string
-    {
-        return trans_choice('strings.users', 1);
-    }
```
Comment (Miniontoby) [inline app/Filament/Resources/UserResource.php]: Again here: Users is not panel related word!

File: resources/views/filament/pages/dashboard.blade.php [hunk 18]
```diff
@@ -6,28 +6,28 @@
...
-            {{ trans_choice('strings.eggs', 2) }}
+            Eggs
             <x-slot name="badge">{{ $eggsCount }}</x-slot>
         </x-filament::tabs.item>
 
         <x-filament::tabs.item
             icon="tabler-users"
         >
-            {{ trans_choice('strings.users', 2) }}
+            Users
```
Comment (Miniontoby) [inline resources/views/filament/pages/dashboard.blade.php]: And here, yeah. Why untranslate Users?

**Comment by Miniontoby** [comment]
> Oh I see, parts of it are already merged...

Or maybe not... hmmm

**Comment by Miniontoby** [comment]
I do not know if I want to keep this open, since literally 90% of my changes is removed!

File: app/Filament/Resources/UserResource.php (hunk 17 above)
Comment (lancepioch) [reply app/Filament/Resources/UserResource.php]: Yes, Users are a resource. Resources don't make sense to be translated.

File: resources/views/filament/pages/dashboard.blade.php (hunk 18 above)
Comment (lancepioch) [reply resources/views/filament/pages/dashboard.blade.php]: Yes, Users are a resource. Resources don't make sense to be translated.

**Comment by lancepioch** [comment]
Thanks for your changes.
